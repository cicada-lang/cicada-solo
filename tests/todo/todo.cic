a: Nat = TODO("A trivial Nat")

b: Trivial = TODO("It's not Trivial")

c: Absurd = TODO("Absurd!")

law_of_excluded_middle(A: Type): Either(A, (A) -> Absurd) {
  TODO("tertium non datur")
}

a
b
c
law_of_excluded_middle
