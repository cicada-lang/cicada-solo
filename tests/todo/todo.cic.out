@TODO "A trivial Nat": Nat
sole: Trivial
the(Absurd, @TODO "Absurd!"): Absurd
(A) => { @TODO "tertium non datur" }: (A: Type) -> Either(A, (_: A) -> Absurd)
